blockchain/pallets/token_distribution/src/lib.rs

```diff
@@ -0,0 +1,172 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub use pallet::*;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+pub mod exponential_issuance;
+pub use exponential_issuance::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+    use codec::{Decode, Encode};
+
+    use frame_support::pallet_prelude::*;
+    use frame_system::pallet_prelude::*;
+
+    use frame_support::traits::{Currency, Get};
+
+    pub type FractalId = u64;
+
+    type BalanceOf<T> =
+        <<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+
+        type Currency: Currency<Self::AccountId>;
+
+        type TotalIssuance: Get<BalanceOf<Self>>;
+        type IssuanceHalfLife: Get<Self::BlockNumber>;
+        type IssuanceCompleteAt: Get<Self::BlockNumber>;
+    }
+
+    #[pallet::storage]
+    pub type TotalIssuanceOffset<T: Config> = StorageValue<_, BalanceOf<T>, ValueQuery>;
+
+    #[pallet::storage]
+    pub type DestinationWeights<T: Config> =
+        StorageMap<_, Blake2_128Concat, Destination<T::AccountId>, u32, ValueQuery>;
+
+    #[pallet::storage]
+    pub type PurposeBalances<T: Config> =
+        StorageMap<_, Blake2_128Concat, u8, BalanceOf<T>, ValueQuery>;
+
+    #[pallet::pallet]
+    #[pallet::generate_store(pub(super) trait Store)]
+    pub struct Pallet<T>(_);
+
+    #[pallet::event]
+    #[pallet::metadata(BalanceOf<T> = "Balance")]
+    pub enum Event<T: Config> {}
+
+    #[pallet::error]
+    pub enum Error<T> {}
+
+    #[derive(Debug, Clone, PartialEq, Eq, Encode, Decode)]
+    pub enum Destination<A> {
+        Address(A),
+        Purpose(u8),
+    }
+
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        #[pallet::weight((
+            10_000 + T::DbWeight::get().reads_writes(0, 1),
+            DispatchClass::Normal,
+            Pays::No
```

Comment:
Rust's formatter `rustfmt` is ambivalent about having this trailing comma. The normal reason for including these trailing commas are for simpler diffs when adding values. Since these are all functions, it's highly unlikely they'll get more lines, thus the diff benefit is very small. I'd prefer to not have this specified our style guide and therefore allow either.

So unless you have a strong opposition, I'm leaving these as is.

Reply:
I don't see an advantage to not being consistent, but it's not a strong opposition